e2e_test/batch/types/arithmetic.slt.part

```diff
@@ -71,4 +71,55 @@ select interval '12 days' / 4.2;
 query T
 SELECT interval '1 month' / 2000;
 ----
-00:21:36
\ No newline at end of file
+00:21:36
+
+query T
+select Date '2022-06-23' + 4;
+----
+2022-06-27
+
+query T
+select 4 + Date '2022-06-23';
+----
+2022-06-27
+
+query T
+select Date '2022-06-23' - 4;
+----
+2022-06-19
+
+query T
+select Date '2022-06-23' + TIME '19:24:00';
+----
+2022-06-23 19:24:00
+
+query T
+select TIME '19:24:00' +  Date '2022-06-23';
+----
+2022-06-23 19:24:00
+
+query T
+select TIME '18:20:49' -  TIME '16:07:16';
+----
+02:13:33
+
+query T
+select TIME '18:20:49' -  interval '2' hour;
+----
+16:20:49
+
+query T
+select TIME '18:20:49' -  interval '1' days;
+----
+18:20:48
+
+query T
+select TIME '18:20:49' +  interval '1' days;
+----
+18:20:50
```

Comment:
These cases are very confusing. The first case is actually parsed as
```
select (TIME '18:20:49' -  interval '1') AS "days";
----
```
Where `interval '1'` uses the default unit `second` and `"days"` is the alias of output column.

In PostgreSQL, when unit is outside single quotes, it has to be singular; when inside single quotes, it can be either singular or plural.

Reply:
I misunderstood the use of interval😰